make program cooler by using terminal graphic display

StephenHogg opened the pull request:
made the visuals cool by using ncurses. Only changed 2 functions, but haven't changed any of your tests!!!! ohohoh

Note that it's only displaying the first 20 or so rows at the moment - if you try and display more rows than you have in your terminal then ncurses will error and terminate the program. Suggested solution: get big screen

StephenHogg committed:
made the visuals cool by using ncurses. Only changed 2 functions, but haven't changed any of your tests!!!! ohohoh

tomlockwood commented on lib/game.py:
```diff
@@ -22,7 +23,7 @@ def __init__(self,rules=Rules(),grid=Grid(),period_retention=0):
     for x in range(self.grid.x):
       for y in range(self.grid.y):
-        self.grid.state[x,y] = random.choice(range(self.rules.amount))
+        self.grid.state[x][y] = random.choice(range(self.rules.amount))
```
I was under the impression the other way was more in keeping with numpy ndarrays?

StephenHogg replied:
mate it's your change lol. Just got incorporated into my code somehow because of a commit that was made while I was making this one.

tomlockwood replied:
ohohoho mate learn to use `git merge master`

tomlockwood commented on lib/game.py:
```diff
@@ -69,23 +70,24 @@ def periodic(self):
-    print (WHITE_TEXT + 'At tick number ' + str(self.ticks))
+    i = 0
+    j = 0
+    self.stdscr.addstr(i, 0, 'At tick number ' + str(self.ticks), curses.color_pair(1))
```
DISRESPECTS the rgb in the rule.colour dict, but i will fix!

StephenHogg replied:
consider urself disrespected ohohoh. Also change the things in `color_pair` in `load_game` to fix this easily. Can just init a bunch of pairs from your dict.

tomlockwood commented on load_game.py:
```diff
@@ -1,31 +1,43 @@
+
+      print('r=replay, rg=random grid replay, f=change filename')
+      print('enter nothing for next')
+      opt = input('Option?: ')
```
this dun work under curses

StephenHogg replied:
:<

StephenHogg committed:
Now it just shows as many rows as can be fit in the screen!! Any extras get rendered outside the window

don't make the screen smaller after you start it running though that's bad

ohoho
